server/ocsp.go

```diff
@@ -400,15 +409,248 @@ func (srv *Server) NewOCSPMonitor(tc *tls.Config) (*tls.Config, *OCSPMonitor, er
 			}, nil
 		}
 
-		// GetClientCertificate returns a certificate that's presented to a
-		// server.
-		tc.GetClientCertificate = func(info *tls.CertificateRequestInfo) (*tls.Certificate, error) {
-			return &cert, nil
+		// Check whether need to verify staples from a client connection depending on the type.
+		switch kind {
+		case typeStringMap[ROUTER], typeStringMap[GATEWAY], typeStringMap[LEAF]:
+			tc.VerifyConnection = func(s tls.ConnectionState) error {
+				oresp := s.OCSPResponse
+				if oresp == nil {
+					return fmt.Errorf("%s client missing OCSP Staple", kind)
+				}
+
+				// Client route connections will verify the response of the staple.
+				if len(s.VerifiedChains) == 0 {
+					return fmt.Errorf("%s client missing TLS verified chains", kind)
+				}
+				chain := s.VerifiedChains[0]
+				resp, err := ocsp.ParseResponseForCert(oresp, chain[0], issuer)
```

Comment:
Just quick note here, the `issuer` here in the callback is already the result `getOCSPIssuer` so no need to call it once again.